adafruit_sht31d.py

```diff
@@ -121,7 +128,7 @@ def _crc(data):
     return crc & 0xFF
 
 
-def _unpack(data):
+def _unpack(data) -> float:
```

Comment:
Looking at the return, it definitely returns a `List`, and looking in the code it looks like it starts out as `None` and then goes over and rewrites them to `int` (based on the `struct.unpack`).  I might be getting a little lost in the math, but it does seem like it completely rewrites all values, and everything using this function makes no attempt to verify if there are any `None` values in the list before using them like `int`s.  Therefore, this should be type annotated with a return of `List[int]`.

We can also type annotate the `data` argument.  Based on how it's used, let's do `from circuitpython_typing import ReadableBuffer` in the typing import block and annotate it here as `ReadableBuffer`.  That's because it's best (typically, but not always) to annotate arguments as *generally* as possible, and annotate returns as *strictly* as possible.  The `List[int]` is the strictest definition of the return, and `ReadableBuffer` is the most general definition allowed for the input!